proposal: optionally support lb-ttsc, for ttypescript

mschnee opened the pull request:
This PR provides two things:
- :heavy_plus_sign: Fixes the `.husky/pre-commit: 4: [: unexpected operator` error on commit
- :heavy_plus_sign: Provides an optional `lb-ttsc` entry to allow consumers to optionally use `ttypescript` (to support things like `ts-auto-mock` in testing)

Signed-off-by: Matt <i.am@matthew.engineer>


### Checklist

- [ ] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [ ] `npm test` passes on your machine
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated

raymondfeng commented:
@mschnee Any pointer to `ttypescript`?

mschnee commented:
@raymondfeng pointer, as in documentation?  Or a dependency reference?  I'm still thinking about how best to word documentation (and where to put it).  I am not sure at the moment that `ttypescript` should be listed as a dependency, or even an optional dependency.

[ttypescript](https://github.com/cevek/ttypescript) is a "wrapper around typescript's APIs to support pluggable transformers", and is a requirement for things like [ts-auto-mock](https://github.com/Typescript-TDD/ts-auto-mock)

raymondfeng commented:
> @raymondfeng pointer, as in documentation? Or a dependency reference? I'm still thinking about how best to word documentation (and where to put it). I am not sure at the moment that `ttypescript` should be listed as a dependency, or even an optional dependency.
> 
> [ttypescript](https://github.com/cevek/ttypescript) is a "wrapper around typescript's APIs to support pluggable transformers", and is a requirement for things like [ts-auto-mock](https://github.com/Typescript-TDD/ts-auto-mock)

Yes, the github link helps. `ttypescript` is cool as I have been wondering about the possibility to reflect generic types. Maybe we can just add a flag to `lb-tsc` to enable usage of `ttypescript`?

mschnee commented:
I'm not opposed to a flag, though it is inconsistent with how users of ttypescript expectations (which is why I used `lb-ttsc`).

mschnee commented:
I've gone ahead and also added an option, `--use-ttypescript`, and have updated `compile-package.js` to check, if `--use-typescript` or `lb-ttsc` is invoked without `ttypescript` being installed, also adding a unit test to cover the case where the option is invoked but the dependency is not installed.

I've also added some documentation to the `README` noting that `ttypescript` support is available but experimental.
